File: exp-6 des algorithm.py
from Crypto.Cipher import DES
from Crypto.Util.Padding import pad, unpad

key = input("Enter 8-character key: ").encode()
choice = input("Enter E to Encrypt or D to Decrypt: ")

des = DES.new(key, DES.MODE_ECB)

if choice.upper() == 'E':
    text = input("Enter plain text: ").encode()
    cipher = des.encrypt(pad(text, 8))
    print("Encrypted Text:", cipher)

elif choice.upper() == 'D':
    cipher = eval(input("Enter encrypted text: "))
    plain = unpad(des.decrypt(cipher), 8)
    print("Decrypted Text:", plain.decode())

else:
    print("Invalid choice")

File: exp 6- des algorithm.cpp
#include <stdio.h>
#include <string.h>

void des_process(char text[], int key)
{
    int i;

    for(i = 0; i < strlen(text); i++)
    {
        text[i] = text[i] ^ key;   // XOR operation
    }
}

int main()
{
    char text[100];
    int key;
    char choice;

    printf("Enter E for Encryption or D for Decryption: ");
    scanf(" %c", &choice);

    printf("Enter the text: ");
    scanf("%s", text);

    printf("Enter the key: ");
    scanf("%d", &key);

    if(choice == 'E' || choice == 'e')
    {
        des_process(text, key);
        printf("Encrypted text: %s\n", text);
    }
    else if(choice == 'D' || choice == 'd')
    {
        des_process(text, key);
        printf("Decrypted text: %s\n", text);
    }
    else
    {
        printf("Invalid choice\n");
    }

    return 0;
}